### ISSUE TITLE
SwayFX does not allow non-primary GPUs to suspend

### ISSUE THREAD

**Issue by Sid127** [WillPower3309/swayfx#274]
### Please fill out the following:
- **Swayfx Version: 0.3.2**

- **Debug Log: https://paste.sidonthe.net/raw/bear-bear-gecko**

- **Configuration File: default configuration**

- **Description:**
    - own a multi-gpu system, such as a gaming laptop
    - launch swayfx

Swayfx currently initializes its DRM backend for every card available on the system, as can be seen in the log I have attached. Yes, I know I'm on the proprietary drivers, but I have also reproduced this using nouveau. This causes the non-primary GPU to stay active, preventing it from engaging in a runtime power management routine (entering D3cold state), increasing power consumption.

**Comment by WillPower3309** [comment]
Thanks for raising the issue! I assume this isn't the case for upstream sway, but can you confirm?

**Comment by Sid127** [comment]
Yes, this is not an issue with upstream, whichever is the current version of sway in the arch repos. It is, however, also an issue on hyprland-git

**Comment by swwwee** [comment]
I have the same issue with regular sway-1.9, using a 3060M. It is most likely an underlying issue with the nvidia-drivers as a whole, or wlroots not wanting to playing nice (both are possible).

E.g., I have my main monitor plugged into the iGPU to get around nauseating, terrible flickering. And I use reverse-prime, to offload GPU intensive tasks. In theory, the iGPU should only be used, and it is, until a process is offloaded. This powers on the GPU, but it also moves sway onto the graphics stack, preventing the GPU from powering off. 

It really sucks, but we cannot control NVIDIA drivers or the wlroots project's stance on said drivers.

**Comment by WillPower3309** [comment]
> Yes, this is not an issue with upstream, whichever is the current version of sway in the arch repos. It is, however, also an issue on hyprland-git

To clarify, is this sway-git or was it 1.9?

**Comment by Sid127** [comment]
it was whatever the latest release of Sway was at the time.

-------- Original Message --------
